rbx_dom_weak/src/brick_color.rs

```diff
@@ -0,0 +1,176 @@
+use serde_derive::{Serialize, Deserialize};
+use std::{convert::TryFrom, fmt};
+
+macro_rules! make_brick_color {
+	({$([$enum:ident, $name:tt, $value:tt],)+}) => {
+		#[derive(Debug, Clone, Copy, PartialEq, Serialize, Deserialize)]
+		pub enum BrickColor {
+			$($enum,)+
+		}
+
+		// TODO: These should work for all numeric types, but making a macro was too difficult
+		impl TryFrom<u8> for BrickColor {
+			type Error = ();
+
+			fn try_from(value: u8) -> Result<Self, Self::Error> {
+				match value {
+					$(
+						$value => Ok(BrickColor::$enum),
+					)+
+
+					_ => Err(()),
+				}
+			}
+		}

```

Comment:
I'm not super comfortable bumping the minimum Rust version for `TryFrom`, can we just use an inherent impl with a descriptive name?

Reply:
Done.